Hackney.Core/Middleware/Exception/ExceptionMiddlewareExtensions.cs

```diff
@@ -0,0 +1,54 @@
+﻿using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Threading.Tasks;
+
+[assembly: InternalsVisibleTo("Hackney.Core.Tests")]
```

Comment:
Perhaps my only objection - I was never a fan of marking something internal, then being unable to test it and using this. (but it's an opinion more than anything, so here's an approval!)

Reply:
I hear you.
I only did it because I had to remove the functional bit into a separate function just so that it could be tested. Ideally that function would be private (it definitely shouldn't be public in a public-facing nuget package) but I needed a way of being able to unit test it, hence making it an internal function and adding this line.